libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```
@@ -0,0 +1,207 @@
+                    if (string.IsNullOrEmpty(body) || request.Streams?.Count == 0)
+                    {
+                        response.StatusCode = (int)HttpStatusCode.BadRequest;
+                        logger?.LogInformation("Request missing body and/or streams.");
+                        return response;
+                    }
+
+                    try
+                    {
+                        var adapter = new BotFrameworkStreamingExtensionsAdapter(Server, middlewareSet, logger);
+                        var bot = services?.GetService<IBot>() ?? this.bot;
+
+                        if (bot == null)
+                        {
+                            throw new Exception("Unable to find bot when processing request.");
+                        }
+
+                        if (onTurnError != null)
+                        {
+                            adapter.OnTurnError = onTurnError;
+                        }
+
+                        var invokeResponse = await adapter.ProcessActivityAsync(body, request.Streams, new BotCallbackHandler(bot.OnTurnAsync), CancellationToken.None).ConfigureAwait(false);
+
+                        if (invokeResponse == null)
+                        {
+                            response.StatusCode = (int)HttpStatusCode.OK;
+                        }
+                        else
+                        {
+                            response.StatusCode = invokeResponse.Status;
+                            if (invokeResponse.Body != null)
+                            {
+                                response.SetBody(invokeResponse.Body);
+                            }
+                        }
+
+                        invokeResponse = null;
+                    }
+                    catch (Exception ex)
+                    {
+                        response.StatusCode = (int)HttpStatusCode.InternalServerError;
+                        logger?.LogError(ex.Message);
+                    }
```

More null checkes. Ick! 

Why wouldn't you let this bubble up?